js/script.js

```diff
@@ -0,0 +1,47 @@
+const url = 'https://restcountries.eu/rest/v2/name/';
+const countriesList = document.getElementById('countries');
+document.getElementById('search').addEventListener('click', searchCountries);
+
+function searchCountries() {
+    let countryName = document.getElementById('country-name').value;
+    if (!countryName.length) countryName = 'Poland';
+    fetch(url + countryName)
+        .then(function (resp) {
+           return resp.json();
+        })
+        .then(showCountriesList)
+}
+
+function generateTemplate(name, data, basicElement) {
+    const template = document.getElementById(name).innerHTML;
+    let element = document.createElement(basicElement || 'li');
+
+    Mustache.parse(template);
+    element.innerHTML = Mustache.render(template, data);
+
+    return element;
+}
+
+function showCountriesList(resp) {
+    countriesList.innerHTML = '';
```

Comment:
this method of cleaning DOM element is good, but it doesn't work e.g on IE

try to find different method for cleaning DOM element, tip: you can use while loop